pkg/api/chunk.go

```diff
@@ -68,22 +62,38 @@ func (s *server) chunkUploadHandler(w http.ResponseWriter, r *http.Request) {
 		if jsonhttp.HandleBodyReadError(err, w) {
 			return
 		}
-		s.Logger.Debugf("chunk upload: read chunk data error: %v, addr %s", err, address)
+		s.Logger.Debugf("chunk upload: read chunk data error: %v", err)
 		s.Logger.Error("chunk upload: read chunk data error")
 		jsonhttp.InternalServerError(w, "cannot read chunk data")
 		return
 	}
 
-	chunk := swarm.NewChunk(address, data)
-	if !content.Valid(chunk) {
-		if !soc.Valid(chunk) {
-			s.Logger.Debugf("chunk upload: invalid chunk: %s", address)
-			s.Logger.Error("chunk upload: invalid chunk")
-			jsonhttp.BadRequest(w, nil)
-			return
-		}
+	if len(data) < swarm.SpanSize {
```

Comment:
This would allow zero payload length chunks when the span is all zero. Before this change this was not allowed, minimum payload size was 1 (therefore the minimum data size was `SpanSize + 1`). I don't have a strong opinion about this just wanted to mention.

Reply:
yeah this is indeed a problem. It is not clearly defined whether a chunk with span `0` and `nil` data is a valid chunk or not

Reply:
no the span can be arbitrary metadata really, it only matters for swarm file chunks.
if the query  paramater is not given, it falls back to datalength by  default

Reply:
> yeah this is indeed a problem. It is not clearly defined whether a chunk with span `0` and `nil` data is a valid chunk or not

true, I think it is valid according to definition but i would not allow 0  length content